crates/client/l1-messages/src/contract.rs

```diff
@@ -19,35 +22,46 @@ pub enum L1EventToTransactionError {
 
 pub fn parse_handle_l1_message_transaction(
     event: LogMessageToL2Filter,
-) -> Result<HandleL1MessageTransaction, L1EventToTransactionError> {
+) -> Result<L1HandlerTransaction, L1EventToTransactionError> {
     // L1 from address.
     let from_address = Felt252Wrapper::try_from(sp_core::U256::from_big_endian(event.from_address.as_bytes()))
-        .map_err(L1EventToTransactionError::InvalidFromAddress)?;
+        .map_err(L1EventToTransactionError::InvalidFromAddress)?
+        .into();
 
     // L2 contract to call.
     let contract_address = Felt252Wrapper::try_from(sp_core::U256(event.to_address.0))
-        .map_err(L1EventToTransactionError::InvalidContractAddress)?;
+        .map_err(L1EventToTransactionError::InvalidContractAddress)?
+        .into();
 
     // Function of the contract to call.
     let entry_point_selector = Felt252Wrapper::try_from(sp_core::U256(event.selector.0))
-        .map_err(L1EventToTransactionError::InvalidEntryPointSelector)?;
+        .map_err(L1EventToTransactionError::InvalidEntryPointSelector)?
+        .into();
 
     // L1 message nonce.
-    let nonce: u64 = Felt252Wrapper::try_from(sp_core::U256(event.nonce.0))
-        .map_err(L1EventToTransactionError::InvalidNonce)?
-        .try_into()
-        .map_err(L1EventToTransactionError::InvalidNonce)?;
+    let nonce =
+        Felt252Wrapper::try_from(sp_core::U256(event.nonce.0)).map_err(L1EventToTransactionError::InvalidNonce)?.into();
 
-    let event_payload: Vec<Felt252Wrapper> = event
+    let event_payload: Vec<_> = event
         .payload
         .iter()
-        .map(|param| Felt252Wrapper::try_from(sp_core::U256(param.0)))
-        .collect::<Result<Vec<Felt252Wrapper>, Felt252WrapperError>>()
+        .map(|param| Felt252Wrapper::try_from(sp_core::U256(param.0)).map(StarkFelt::from))
+        .collect::<Result<Vec<_>, Felt252WrapperError>>()
         .map_err(L1EventToTransactionError::InvalidCalldata)?;
 
-    let mut calldata: Vec<Felt252Wrapper> = Vec::with_capacity(event.payload.len() + 1);
-    calldata.push(from_address);
-    event_payload.iter().collect_into(&mut calldata);
+    let calldata = {
+        let mut calldata: Vec<_> = Vec::with_capacity(event.payload.len() + 1);
+        calldata.push(from_address);
+        event_payload.iter().collect_into(&mut calldata);
+
+        Calldata(Arc::new(calldata))
+    };
 
-    Ok(HandleL1MessageTransaction { nonce, contract_address, entry_point_selector, calldata })
+    Ok(L1HandlerTransaction {
+        nonce,
+        contract_address,
+        entry_point_selector,
+        calldata,
+        version: TransactionVersion(StarkFelt::ZERO),
```

Comment:
what does this correspond to?

Reply:
It is the version of the current (and only) version of the handle l1 message transaction